src/compas_viewer/scene/geometryobject.py

```diff
@@ -0,0 +1,124 @@
+from typing import Optional
+from typing import Tuple
+
+from compas.colors import Color
+from compas.geometry import Geometry
+from compas.geometry import Line
+from compas.geometry import Point
+from compas.scene import GeometryObject as BaseGeometryObject
+
+from .sceneobject import ShaderDataType
+from .sceneobject import ViewerSceneObject
+
+
+class GeometryObject(ViewerSceneObject, BaseGeometryObject):
+    """Viewer scene object for displaying COMPAS Geometry.
+
+    Parameters
+    ----------
+    geometry : :class:`compas.geometry.Geometry`
+        A COMPAS geometry.
+    pointcolor : :class:`compas.colors.Color`, optional
+        The color of the points. Global settings in the viewer will be used if not specified.
+    linecolor : :class:`compas.colors.Color`, optional
+        The color of the lines. Global settings in the viewer will be used if not specified.
+    surfacecolor : :class:`compas.colors.Color`, optional
+        The color of the surfaces. Global settings in the viewer will be used if not specified.
+    **kwargs : dict, optional
+        Additional options for the :class:`compas_viewer.scene.ViewerSceneObject`
+        and :class:`compas.scene.GeometryObject`.
+
+    Attributes
+    ----------
+    geometry : :class:`compas.geometry.Geometry`
+        The COMPAS geometry.
+    pointcolor : :class:`compas.colors.Color`
+        The color of the points.
+    linecolor : :class:`compas.colors.Color`
+        The color of the lines.
+    surfacecolor : :class:`compas.colors.Color`
+        The color of the surfaces.
+    mesh : :class:`compas.datastructures.Mesh`
+        The triangulated mesh representation of the geometry.
+    LINEARDEFLECTION : float
+        The default linear deflection for the geometry.
+
+    See Also
+    --------
+    :class:`compas.geometry.Geometry`
+    """
+
+    LINEARDEFLECTION = 0.2
```

Comment:
Yes, we talked about this and this was the temporary solution at the time. The problem is, currently `compas.tolerance.TOL.lineardeflection` =1e-3 which is too small and super detailed. should I modify the number?

Reply:
i think you are using this value in the wrong way. `u` and `v` parameters for the discretisation of `compas.geometry` shapes should be configurable by the user, with a sensible default like `u=64` and `v=u`. linear deflection is used when discretising Nurbs and Brep. there it depends on the size of the model. since most people tend to work in meters, 1e-3 makes sense. if they work in mm they can set the linear deflection to 1 or 0.1 ...